### ISSUE TITLE
Research how to test module semantics

### ISSUE THREAD

**Issue by brianloveswords** [bocoup/test262#48]

**Comment by jugglinmike** [comment]
I've been looking into this over the past few days. I'm still exploring the
surface area here, but at a high-level, I think I've identified three distinct
aspects of this work:
- `import`/`export` declarations (this represents the majority of the work and
  probably the most critical functionality)
- "module code"
- the module namespace exotic object

All this will require some formal specification of module resolution behavior.
I have an idea for a minimal-viable solution that should be compatible with
most general-purpose loaders (including [the whatwg loader](https://github.com/whatwg/loader)), but we'll need
a place to document that!

To lay the groundwork, I've proposed that Test262 maintain a `INTERPRETING.md`
file to formalize the expectations for consumers who are executing the tests in
their runtimes. See https://github.com/tc39/test262/pull/508

In the mean time, I will continue my research outlining what needs to be done.
I expect to resolve this issue by creating 3 (or more) more directed issues
describing concrete test-writing tasks.
